### PR TITLE
Add venue management

### LINKED ISSUE #1324: Add venue management
This will eventually allow creators of event to pin location on map. We can reverse geocode the location from coordinates to get the address. Display the address for editing if creator wants to tweak something in it. Store the address with coordinates.

Proposed solution: 
1. Use Google Maps API to display map. The Google Maps SDK for Android and iOS is free.
2. Use [google_maps_flutter](https://pub.dev/packages/google_maps_flutter) package to display map screen and set markers on location.
3. Utilize [geocoding](https://pub.dev/packages/geocoding) package to reverse geocode the location and get address.
4. Utilize [geolocator](https://pub.dev/packages/geolocator) package to get current location of creator.
5. Pass coordinates to backend for possible future use.

### PR DESCRIPTION
**Pull request by guptapriyanshu7**
Added google maps api for venue selection
Display a map and mark location
Reverse geocode from coordinates
Store coordinates in database

Fixes #1324

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
Priyanshu, please create an issue and make it autoclose with this PR. We need to stick with the procedure. That way others, including your mentors can comment prior to your submital of the PR.

**Comment by guptapriyanshu7** [comment]
Hey @palisadoes, opened a new issue.

File: lib/views/after_auth_screens/events/create_event_form.dart [hunk 1]
```diff
@@ -70,6 +85,26 @@ class CreateEventForm extends StatelessWidget {
...
+          //       child: Text(
+          //         AppLocalizations.of(context)!.strictTranslate('Add Location'),
+          //         style: Theme.of(context)
+          //             .textTheme
+          //             .headline5!
+          //             .copyWith(fontSize: 16),
+          //       ),
+          //     )
+          //   ],
+          // ),
```
Comment (JamaicanFriedChicken) [inline lib/views/after_auth_screens/events/create_event_form.dart:107]: May I ask why is this commented out for now?

**Review by JamaicanFriedChicken — CHANGES_REQUESTED** [review]
Also @guptapriyanshu7, the testing build for android is failing for this PR. Please resolve~

File: lib/views/after_auth_screens/events/create_event_form.dart (hunk 1 above)
Comment (guptapriyanshu7) [reply lib/views/after_auth_screens/events/create_event_form.dart:107]: ok i forgot to remove this.

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by guptapriyanshu7** [comment]
Hey @palisadoes, can you make this change in github actions file, https://stackoverflow.com/questions/71924098/flutter-github-actions-bad-cpu-type-in-executable. I am not authorized.

**Comment by palisadoes** [comment]
Shouldn't we be testing for other hardware types? This specifies ARM only. Are there other architectures we should add, and if so, what should they be? Please coordinate this with your mentor.

**Comment by palisadoes** [comment]
Will this change allow the code to run on all iOS platforms? I ask because it doesn't make sense to correct the error without knowing whether all potential users will be able to use the code.

Are you proposing that we eventually run two workflows, one for arm and the other for x86?

**Review by JamaicanFriedChicken — APPROVED** [review]
Looks good

**Review by palisadoes — APPROVED** [review]

**Pull request merged**